plugins/out_s3/s3_parquet.c

```diff
@@ -0,0 +1,705 @@
+        ret = -1;
+        goto error;
+    }
+
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -parquetPageSize ", 18);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    tmp = flb_sds_printf(&amount_page, "%zu", ctx->parquet_page_size);
+    if (!tmp) {
+        flb_errno();
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              amount_page, strlen(amount_page));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -parquetRowGroupSize ", 22);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    tmp = flb_sds_printf(&amount_row_group, "%zu", ctx->parquet_row_group_size);
+    if (!tmp) {
+        flb_errno();
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              amount_row_group, strlen(amount_row_group));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -recordType ", 13);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              ctx->parquet_record_type,
+                              flb_sds_len(ctx->parquet_record_type));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -schemaType ", 13);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              ctx->parquet_schema_type,
+                              flb_sds_len(ctx->parquet_schema_type));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -schemaFile ", 13);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              ctx->parquet_schema_file,
+                              flb_sds_len(ctx->parquet_schema_file));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              " -output ", 9);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf,
+                              outfile, strlen(outfile));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf, " ", 1);
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    result = flb_sds_cat_safe(cmd_buf, infile, strlen(infile));
+    if (result < 0) {
+        ret = -1;
+        goto error;
+    }
+
+    flb_sds_destroy(amount_page);
+    flb_sds_destroy(amount_row_group);
+
+    return 0;
+
+error:
```

Comment:
Should we consider rename to cleanup? Then on Line196 you can set ret = 0. This control block cleans up the dangling items and returns the error code.